vita3k/modules/SceHttp/SceHttp.cpp

```diff
@@ -145,57 +522,584 @@ EXPORT(int, sceHttpGetIcmOption) {
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = emuenv.http.requests.find(reqId);
+
+    // These methods have no body
+    if (req->second.method == SCE_HTTP_METHOD_HEAD || req->second.method == SCE_HTTP_METHOD_OPTIONS)
+        return 0;
+
+    auto conn = emuenv.http.connections.find(req->second.connId);
+    auto tmpl = emuenv.http.templates.find(conn->second.tmplId);
+
+    SceSize read = 0;
+
+    // If the game wants to read more than whats available, change the read ammount to what is available
+    if (size > (req->second.res.contentLength - req->second.res.responseRead)) {
+        size = req->second.res.contentLength - req->second.res.responseRead;
+    }
+
+    if (req->second.res.responseRead == req->second.res.contentLength) {
+        // If we already have read all the response.
+        return 0;
+    }
+
+    memcpy(data, req->second.res.body + req->second.res.responseRead, size);
+
+    req->second.res.responseRead += read;
+
+    return read;
 }
 
-EXPORT(int, sceHttpRequestGetAllHeaders) {
+EXPORT(int, sceHttpRedirectCacheFlush) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpSendRequest) {
-    return UNIMPLEMENTED();
+EXPORT(SceInt, sceHttpRemoveRequestHeader, SceInt reqId, const char *name) {
+    if (!emuenv.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (emuenv.http.requests.find(reqId) == emuenv.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = emuenv.http.requests.find(reqId);
+
+    if (req->second.headers.find(name) != req->second.headers.end())
+        req->second.headers.erase(name);
+
+    return 0;
+}
+
+EXPORT(SceInt, sceHttpRequestGetAllHeaders, SceInt reqId, Ptr<char> *header, SceSize *headerSize) {
+    if (!emuenv.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (emuenv.http.requests.find(reqId) == emuenv.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = emuenv.http.requests.find(reqId);
+
+    auto headers = net_utils::constructHeaders(req->second.headers);
+
+    auto h = Ptr<char>(alloc(emuenv.mem, sizeof(char), "headers"));
+    memcpy(h.get(emuenv.mem), headers.data(), headers.length() + 1);
+    req->second.guestPointers.push_back(h);
+    *header = h;
+
+    *headerSize = headers.length() + 1;
+
+    return 0;
+}
+
+EXPORT(SceInt, sceHttpSendRequest, SceInt reqId, const char *postData, SceSize size) {
+    if (!emuenv.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (emuenv.http.requests.find(reqId) == emuenv.http.requests.end())
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_ID);
+
+    auto req = emuenv.http.requests.find(reqId);
+    auto conn = emuenv.http.connections.find(req->second.connId);
+    auto tmpl = emuenv.http.templates.find(conn->second.tmplId);
+
+    LOG_DEBUG("Sending {} request to {}", net_utils::int_method_to_char(req->second.method), req->second.url);
```

Comment:
Put it as a LOG_TRACE instead (and same for most of your log_debug)

Reply:
changed some missing cases to warns. The rest to trace